File: backend/app/schemas/log.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


ClickSource = Literal["search", "assistant", "favorite"]


class ClickLogCreate(BaseModel):
    product_id: int = Field(gt=0)
    source: ClickSource


class ClickLogResponse(BaseModel):
    id: int
    user_id: int | None
    product_id: int
    source: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SearchLogRead(BaseModel):
    id: int
    user_id: int | None
    keyword: str | None
    category_id: int | None
    min_price: int | None
    max_price: int | None
    sort: str | None
    page: int
    limit: int
    result_count: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SearchLogListResponse(BaseModel):
    items: list[SearchLogRead]
    page: int
    limit: int
    total: int
    has_next: bool


class ClickLogProduct(BaseModel):
    id: int
    name: str

    model_config = ConfigDict(from_attributes=True)


class ClickLogRead(BaseModel):
    id: int
    user_id: int | None
    product_id: int
    product: ClickLogProduct
    source: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClickLogListResponse(BaseModel):
    items: list[ClickLogRead]
    page: int
    limit: int
    total: int
    has_next: bool

File: frontend/types/log.ts
import type { Product } from "./product";

export type FavoriteItem = {
  id: number;
  product: Product;
  created_at: string;
};

export type FavoriteListResponse = {
  items: FavoriteItem[];
};

export type ClickSource = "search" | "assistant" | "favorite";

export type ClickLogResponse = {
  id: number;
  user_id: number | null;
  product_id: number;
  source: ClickSource;
  created_at: string;
};

export type SearchLog = {
  id: number;
  user_id: number | null;
  keyword: string | null;
  category_id: number | null;
  min_price: number | null;
  max_price: number | null;
  sort: string | null;
  page: number;
  limit: number;
  result_count: number;
  created_at: string;
};

export type SearchLogListResponse = {
  items: SearchLog[];
  page: number;
  limit: number;
  total: number;
  has_next: boolean;
};

export type ClickLog = {
  id: number;
  user_id: number | null;
  product_id: number;
  product: {
    id: number;
    name: string;
  };
  source: ClickSource;
  created_at: string;
};

export type ClickLogListResponse = {
  items: ClickLog[];
  page: number;
  limit: number;
  total: number;
  has_next: boolean;
};
